pypesto/problem.py

```diff
@@ -303,23 +303,30 @@ def get_full_matrix(
         return x_full
 
     def get_reduced_vector(
-            self, x_full: Union[np.ndarray, None]
+            self, x_full: Union[np.ndarray, None],
+            x_indices: Optional[List[int]] = None
     ) -> Union[np.ndarray, None]:
         """
-        Map vector from dim_full to dim, i.e. delete fixed indices.
+        Keep only those elements, which indices are specified in x_indices
+        If x_indices is not provided, delete fixed indices.
 
         Parameters
         ----------
         x_full: array_like, ndim=1
             The vector in dimension dim_full.
+        x_indices:
+            indices of x_full that should remain
```

Comment:
Very good that you added that the full vector is meant! This can always be a little confusing...